addons/account/tests/test_bad_reconciliation.py

```diff
@@ -0,0 +1,268 @@
+            'customer': True,
+        })
+        self.cc_journal = self.env['account.journal'].create({
+            'name': 'CC Payments Journal',
+            'code': 'CC',
+            'type': 'bank',
+            'company_id': self.company.id,
+            'default_debit_account_id': self.cc_bank_account.id,
+            'default_credit_account_id': self.cc_bank_account.id,
+        })
+        # We want to make sure that the company has no rates, that would
+        # screw up the conversions.
+        company_rates = self.env['res.currency.rate'].search(
+            [('currency_id', '=', self.company.currency_id.id)])
+        company_rates.unlink()
+
+        return res
+
+    def _check_account_balance(self, account):
+        balance = sum(self.env['account.move.line'].search(
+            [('account_id', '=', account.id)]).mapped('balance'))
+        return balance
+
+    def test_01(self):
+        """ Implements a successful scenario """
+        ####
+        # Day 1: Invoice Cust/001 to customer (expressed in CC)
+        # Market value of CC (day 1): 1 CC = $0.5
+        # * Dr. 100 CC / $50 - Accounts receivable
+        # * Cr. 100 CC / $50 - Revenue
+        ####
+        self.env['res.currency.rate'].create({
+            'currency_id': self.currency_cc.id,
+            'name': time.strftime('%Y') + '-01-01',
+            'rate': 2,
+        })
+        invoice_cust_001 = self.env['account.invoice'].create({
+            'partner_id': self.customer.id,
+            'account_id': self.receivable_account.id,
+            'type': 'out_invoice',
+            'currency_id': self.currency_cc.id,
+            'company_id': self.company.id,
+            'date_invoice': time.strftime('%Y') + '-01-01',
+        })
+        self.env['account.invoice.line'].create({
+            'product_id': self.env.ref('product.product_product_4').id,
+            'quantity': 1.0,
+            'price_unit': 100.0,
+            'invoice_id': invoice_cust_001.id,
+            'name': 'product that cost 100',
+            'account_id': self.account_revenue.id,
+        })
+        invoice_cust_001.action_invoice_open()
+        self.assertEqual(invoice_cust_001.residual_company_signed, 50.0)
+        aml = invoice_cust_001.move_id.mapped('line_ids').filtered(
+            lambda x: x.account_id == self.account_revenue)
+        self.assertEqual(aml.credit, 50.0)
+        #####
+        # Day 2: Receive payment for half invoice Cust/001 (in CC)
+        # -------------------------------------------------------
+        # Market value of CC (day 2): 1 CC = $0.5
+
+        # Payment transaction:
+        # * Dr. 50 CC / $25 - CC Bank (valued at market price
+        # at the time of receiving the coins)
+        # * Cr. 50 CC / $25 - Accounts Receivable
+        #####
+        self.env['res.currency.rate'].create({
+            'currency_id': self.currency_cc.id,
+            'name': time.strftime('%Y') + '-01-02',
+            'rate': 2,
+        })
```

Comment:
for the sake of Clarity: if the rate didn't change, don't re-create a res.currency.rate